Common/Securities/SecurityCache.cs

```diff
@@ -139,18 +141,30 @@ public void AddData(BaseData data)
 
         private void AddDataImpl(BaseData data, bool cacheByType)
         {
-            var openInterest = data as OpenInterest;
-            if (openInterest != null)
-            {
-                OpenInterest = (long)openInterest.Value;
-                return;
-            }
-
             var tick = data as Tick;
-            if (tick?.TickType == TickType.OpenInterest)
+            if (tick != null)
             {
-                OpenInterest = (long)tick.Value;
-                return;
+                switch (tick.TickType)
+                {
+                    case TickType.Trade:
+                        if (tick.Quantity != 0) Volume = tick.Quantity;
+                        break;
+
+                    case TickType.Quote:
+                        if (tick.BidPrice != 0) BidPrice = tick.BidPrice;
+                        if (tick.BidSize != 0) BidSize = tick.BidSize;
+
+                        if (tick.AskPrice != 0) AskPrice = tick.AskPrice;
+                        if (tick.AskSize != 0) AskSize = tick.AskSize;
+                        break;
+
+                    case TickType.OpenInterest:
+                        OpenInterest = (long)tick.Value;
+                        return;
+                }
+
+                if (tick.Value != 0) Price = tick.Value;
+                _lastData = data;
```

Comment:
I think `_lastData = data;` will be set bellow if appropriate? One thing that's different here is that before if fillFowarding we wouldn't update the `Price/AskSize/..etc`, seems we should respect that behavior